synapsemonitor/monitor.py

```diff
@@ -113,6 +113,47 @@ def _find_modified_entities_file(syn: Synapse, syn_id: str, days: int = 1) -> li
     return []
 
 
+def _traverse(
+    syn: Synapse,
+    synid_root: str,
+    include_types: typing.List = ["folder", "file", "project"],
+) -> list:
+    """Traverse Synapse entity hierarchy to gather all descendant
+    entities of a root entity.
+    Args:
+        syn: Synapse connection
+        synid_root: Synapse ID of root entity.
+        include_types: Must be a list of entity types (ie. [“folder”,”file”])
+            which can be found here:
+            http://docs.synapse.org/rest/org/sagebionetworks/repo/model/EntityType.html
+    Returns:
+        List of descendant Synapse IDs and root Synapse ID
+    """
+
+    synid_desc = []
+
+    # full traverse depends on examining folder entities, even if not requested
+    include_types_mod = include_types.copy()
+    if "folder" not in include_types_mod:
+        include_types_mod.append("folder")
```

Comment:
```suggestion
    include_types_mod = set(include_types)
    include_types_mod.add("folder")
    include_types_mod = list(include_types_mod)
```

Reply:
I forgot that you can add to a set without checking if something exists in it.  It won't add a duplicate.